tests/integration_tests/functional/test_memory_hp.py

```diff
@@ -27,12 +126,162 @@ def test_virtio_mem_detected(uvm_plain_6_1):
             case "start address":
                 assert value == (512 << 30), "start address doesn't match"
             case "region size":
-                assert value == 1024 << 20, "region size doesn't match"
+                assert (
+                    value == hp_config["total_size_mib"] << 20
+                ), "region size doesn't match"
             case "device block size":
-                assert value == 2 << 20, "block size doesn't match"
+                assert (
+                    value == hp_config["block_size_mib"] << 20
+                ), "block size doesn't match"
             case "plugged size":
                 assert value == 0, "plugged size doesn't match"
             case "requested size":
                 assert value == 0, "requested size doesn't match"
             case _:
                 continue
+
+
+def check_memory_usable(uvm):
+    """Allocates memory to verify it's usable (5% margin to avoid OOM-kill)"""
+    mem_available = MeminfoGuest(uvm).get().mem_available.bytes()
+    # number of 64b ints to allocate as 95% of available memory
+    count = mem_available * 95 // 100 // 8
+
+    uvm.ssh.check_output(
+        f"python3 -c 'Q = 0x0123456789abcdef; a = [Q] * {count}; assert all(q == Q for q in a)'"
```

Comment:
feels a bit fishy. Who knows what Pythin will choose to back `0x0123456789abcdef`. Maybe using `mmap` syscall and just touch all the memory whould be more convincing approach?